### ISSUE TITLE
Build failed on latest master (8159585) when using as drop-in replacement for nixpkgs-mozilla

### ISSUE THREAD

**Issue by werner291** [oxalica/rust-overlay#95]
Hello,

this is my shell.nix:

```
{ }:
let
  rustOverlay = import (builtins.fetchTarball "https://github.com/oxalica/rust-overlay/archive/8159585609a772b041cce6019d5c21d240709244.tar.gz");
  pkgs = import <nixpkgs> {
    overlays = [ rustOverlay ];
  };
  rustChannel = pkgs.rustChannelOf { channel = "stable"; };
in pkgs.mkShell {
  nativeBuildInputs = with pkgs; [
    rustChannel.rust
    rustChannel.rust-src
    alsaLib.dev
    pkg-config
    udev
    vscodium
    #xlibsWrapper
    #libXcursor
        xorg.libX11
        xorg.libXcursor
        xorg.libXrandr
        xorg.libXi
        vulkan-loader
        vulkan-tools
        vulkan-headers
        vulkan-validation-layers
  ];


  shellHook = ''
              export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:${
                pkgs.lib.makeLibraryPath [
                  pkgs.vulkan-loader
                ]
              }"'';
}
```

Seems to be some missing dependency to zlib:

```
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/libLLVM-14-rust-1.61.0-stable.so
    libz.so.1 -> not found!
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libtest-efede371ca078f25.so
    libstd-05b39ac0cb4c5688.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-05b39ac0cb4c5688.so
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libLLVM-14-rust-1.61.0-stable.so
    libz.so.1 -> not found!
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-cov
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-cov
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llc
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llc
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-size
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-size
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-as
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-as
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-strip
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-strip
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-ar
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-ar
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-nm
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-nm
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/opt
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/opt
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-readobj
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-readobj
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-dis
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-dis
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-objcopy
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-objcopy
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-objdump
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-objdump
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-lld
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-lld
    libz.so.1 -> not found!
    libLLVM-14-rust-1.61.0-stable.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld64
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld64
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/cargo-fmt
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/cargo-fmt
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/rls
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/rls
    librustc_driver-2c37baf9e5f6ef22.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
    libstd-05b39ac0cb4c5688.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/rust-demangler
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/rust-demangler
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/rustc
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/rustc
    librustc_driver-2c37baf9e5f6ef22.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
    libstd-05b39ac0cb4c5688.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/clippy-driver
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/clippy-driver
    librustc_driver-2c37baf9e5f6ef22.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
    libstd-05b39ac0cb4c5688.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/rustfmt
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/rustfmt
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/rustdoc
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/rustdoc
    libtest-efede371ca078f25.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
    librustc_driver-2c37baf9e5f6ef22.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
    libstd-05b39ac0cb4c5688.so -> found: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting RPATH to: /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/cargo
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/cargo
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/cargo-clippy
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/cargo-clippy
setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/libexec/cargo-credential-1password
searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/libexec/cargo-credential-1password
auto-patchelf: 3 dependencies could not be satisfied
error: auto-patchelf could not satisfy dependency libz.so.1 wanted by /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/libLLVM-14-rust-1.61.0-stable.so
error: auto-patchelf could not satisfy dependency libz.so.1 wanted by /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libLLVM-14-rust-1.61.0-stable.so
error: auto-patchelf could not satisfy dependency libz.so.1 wanted by /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-lld
auto-patchelf failed to find all the required dependencies.
Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
error: builder for '/nix/store/js5vd71pl33z1jwib2y9py58hyscq8jq-rust-1.61.0-x86_64-unknown-linux-gnu.drv' failed with exit code 1;
       last 10 log lines:
       > setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/cargo-clippy
       > searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/bin/cargo-clippy
       > setting interpreter of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/libexec/cargo-credential-1password
       > searching for dependencies of /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/libexec/cargo-credential-1password
       > auto-patchelf: 3 dependencies could not be satisfied
       > error: auto-patchelf could not satisfy dependency libz.so.1 wanted by /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/libLLVM-14-rust-1.61.0-stable.so
       > error: auto-patchelf could not satisfy dependency libz.so.1 wanted by /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libLLVM-14-rust-1.61.0-stable.so
       > error: auto-patchelf could not satisfy dependency libz.so.1 wanted by /nix/store/gljwhm9v6kr4prj9zvc15b1dr702q0px-rust-1.61.0-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-lld
       > auto-patchelf failed to find all the required dependencies.
       > Add the missing dependencies to --libs or use `--ignore-missing="foo.so.1 bar.so etc.so"`.
       For full logs, run 'nix log /nix/store/js5vd71pl33z1jwib2y9py58hyscq8jq-rust-1.61.0-x86_64-unknown-linux-gnu.drv'.
error: 1 dependencies of derivation '/nix/store/0rmyjrfzg3fxdf0x80als6vv25vy2hs3-rust-legacy-1.61.0.drv' failed to build
```

I'm on NixOS 22.05, btw

**Comment by werner291** [comment]
Ok, looks like getting my rust from `pkgs.rustChannelOf { channel = "stable"; };` doesn't work, It builds fine when I use `rust-bin.stable.latest.default`

So it's just a minor incompatibility with nixpkgs-mozilla, nothing really problematic it seems.

**Comment by oxalica** [comment]
Fixed in b6190a6b1a3293f7df30c154a51c65742aa673a9
